Fixed NoSuchMethodException compatibility issue in DonateActivity

ThanosFisherman opened the pull request:
Fixed crash regarding compatibility issue #100

alexstyl commented on android_mobile/src/main/java/com/alexstyl/specialdates/donate/DonateActivity.java:
```diff
@@ -56,7 +57,8 @@ protected void onCreate(Bundle savedInstanceState) {
         final NestedScrollView scrollView = Views.findById(this, R.id.scroll);
 
-        appBarLayout.addOnOffsetChangedListener(new HideStatusBarListener(getWindow()));
+        if (Version.hasLollipop())
```
Wrap the if with brackets please

ThanosFisherman replied:
Alright

alexstyl commented on android_mobile/src/main/java/com/alexstyl/specialdates/donate/DonateActivity.java:
```diff
@@ -81,13 +83,13 @@ public void onClick(View v) {
                 CoordinatorLayout.LayoutParams params = (CoordinatorLayout.LayoutParams) appBarLayout.getLayoutParams();
                 AppBarLayout.Behavior behavior = (AppBarLayout.Behavior) params.getBehavior();
-                behavior.onNestedFling(coordinator, appBarLayout, null, 0, 50, true);
+                if (behavior != null)
```
Can the behavior even be null?

ThanosFisherman replied:
If you run the app on an emulator like I did, donation activity will crash throwing NPE due to behavior being null so I put this null check to at least prevent the app from crashing even though donation activity won't open on an emulator anyways. Let me know how should I proceed with this.
![behaviornull](https://cloud.githubusercontent.com/assets/4888330/25785676/74faac10-338f-11e7-95a5-280e3ac6fc2e.png)

alexstyl replied:
I would rather understand why the behavior is `null` instead of just checking of null. 

Which version was the emulator? Does that happen with multiple versions of Android?

alexstyl replied:
According to this http://stackoverflow.com/a/31182999/1315110 you can set a behavior to the layout. That way you wouldn't have to worry for it being null. Maybe for some versions of Android there is a default behavior set?

ThanosFisherman replied:
Hey, Actually the NPE error is totally misleading. There is nothing wrong with how behavior is initialized. The real cause of this issue, is that the in-app billing is not supported in emulator. So `onDonateException` is called finishing the activity before `behavior` gets initialized thus resulting to NPE on **some** versions of android (e.g api 19). 

So I'd suggest that you merge this pull request as is for now and later on decide how you wanna handle this rare case.

Check out the attached images. First one shows the NPE crash from null `behavior` that occurs without the if-null check. and second image shows the in-app-billing error which however 
doesn't result to a crash after the if-null check has been applied to `behavior`

![npe](https://cloud.githubusercontent.com/assets/4888330/25878863/96da7056-3536-11e7-86b9-3e280c801fe4.png)

![iaperror](https://cloud.githubusercontent.com/assets/4888330/25878871/a33b2f7a-3536-11e7-9fe4-d1881446aa8d.png)

alexstyl replied:
I see. if that is the case, then the null check is not really needed. The user is not going to be navigated to this Activity unless they have Play Services any way (see the the [MainNavigator](https://github.com/alexstyl/Memento-Calendar/blob/develop/android_mobile/src/main/java/com/alexstyl/specialdates/ui/activity/MainNavigator.java#L45) class)

ThanosFisherman replied:
And that brings me to my next comment. In android emulator with google apis installed `hasPlayStoreInstalled()` always returns true so DonationActivity is always launched leading to the following error dominos. So I don't see a better way to handle this for now (if you wanna handle this at all) so it's up to you. It's just a null check that makes the app crash-proof on Android emulators.

alexstyl commented:
@ThanosFisherman I had a better look into this. As it turns out, there is a NPE in case the billing service is not enabled but it is fine to ignore this for now. I'll get it sorted out on a separate branch. For the purpose of testing this, I simply commented out any code in the `DonationService`.

The null check is not needed if you explicitly define a behavior on the layout.
I have set `app:layout_behavior="android.support.design.widget.AppBarLayout$Behavior"` on the `AppBarLayout` of the activity's xml layout and it works wonders. Give it a go

ThanosFisherman commented:
@alexstyl Cool. Null check is indeed not needed when `app:layout_behavior="android.support.design.widget.AppBarLayout$Behavior"` is set So I added it in the xml file and also squashed the changes into a single commit.

alexstyl commented:
Perfect! Let's merge this then! :)

Many thanks for your contribution 🚢
